packages/CodingStyle/src/Rector/ClassMethod/AnnotateThrowables.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\ClassMethod;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Throw_;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * Adds "throws" DocBlock to methods.
+ */
+final class AnnotateThrowables extends AbstractRector
+{
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes() : array
+    {
+        return [Throw_::class];
+    }
+
+    /**
+     * @param Node|Throw_ $node
+     *
+     * @return Node|null
+     */
+    public function refactor(Node $node) : ?Node
+    {
+        if ($this->isThrowableAnnotated($node)) {
+            return null;
+        }
+
+        return $this->annotateMethod($node);
+    }
+
+    /**
+     * @param Throw_ $node
+     *
+     * @return bool
+     */
+    private function isThrowableAnnotated(Throw_ $node) : bool
+    {
+        $method = $node->getAttribute('methodNode');
+        $throwableParts = $node->expr->class->parts;
```

Comment:
```suggestion
        $throwableParts = $this->getName($node->expr->class);
```

Reply:
Not sure if this is helpful: I need all the parts to deal with both imported and not imported exceptions.

Anyway I will examine better the situation once I will have again a working branch.